ui/components/app/edit-gas-display/edit-gas-display.component.js

```diff
@@ -15,17 +17,124 @@ import AdvancedGasControls from '../advanced-gas-controls/advanced-gas-controls.
 
 import { I18nContext } from '../../../contexts/i18n';
 import ActionableMessage from '../../../pages/swaps/actionable-message';
+import { useGasFeeEstimates } from '../../../hooks/useGasFeeEstimates';
+
+import { getShouldShowFiat } from '../../../selectors';
+import { useUserPreferencedCurrency } from '../../../hooks/useUserPreferencedCurrency';
+import { useCurrencyDisplay } from '../../../hooks/useCurrencyDisplay';
+import { PRIMARY, SECONDARY } from '../../../helpers/constants/common';
+import { decGWEIToHexWEI } from '../../../helpers/utils/conversions.util';
 
 export default function EditGasDisplay({
   alwaysShowForm,
   type,
   showEducationButton,
   onEducationClick,
+  defaultEstimateToUse = 'medium',
 }) {
   const t = useContext(I18nContext);
 
+  const { isGasEstimatesLoading, gasFeeEstimates } = useGasFeeEstimates();
+
   const [warning] = useState(null);
+  const [error, setError] = useState(null);
+
   const [showAdvancedForm, setShowAdvancedForm] = useState(false);
+  const [estimateToUse, setEstimateToUse] = useState(defaultEstimateToUse);
+
+  const [maxPriorityFee, setMaxPriorityFee] = useState(
+    gasFeeEstimates?.[estimateToUse]?.suggestedMaxPriorityFeePerGas,
+  );
+  const [maxFee, setMaxFee] = useState(
+    gasFeeEstimates?.[estimateToUse]?.suggestedMaxFeePerGas,
+  );
+  const [maxPriorityFeeError, setMaxPriorityFeeError] = useState(null);
+  const [maxFeeError, setMaxFeeError] = useState(null);
+
+  const [gasLimit, setGasLimit] = useState(21000);
+  const [gasPrice, setGasPrice] = useState(0);
+
+  const prevIsGasEstimatesLoading = useRef(true);
+  useEffect(() => {
+    if (
+      prevIsGasEstimatesLoading.current === true &&
+      isGasEstimatesLoading === false &&
+      estimateToUse
+    ) {
+      setMaxPriorityFee(
+        gasFeeEstimates?.[estimateToUse]?.suggestedMaxPriorityFeePerGas,
+      );
+      setMaxFee(gasFeeEstimates?.[estimateToUse]?.suggestedMaxFeePerGas);
+    }
+  }, [isGasEstimatesLoading, estimateToUse, gasFeeEstimates]);
+
+  // Validation for the maxPriorityFee and maxFee fields
+  useEffect(() => {
+    const isMaxPriorityFeeError =
+      !isGasEstimatesLoading &&
+      maxPriorityFee < gasFeeEstimates?.low?.suggestedMaxPriorityFeePerGas;
+    const isMaxFeeError =
+      !isGasEstimatesLoading &&
+      maxFee < gasFeeEstimates?.low?.suggestedMaxFeePerGas;
+
+    setMaxPriorityFeeError(
+      isMaxPriorityFeeError ? t('editGasMaxPriorityFeeLow') : null,
+    );
+    setMaxFeeError(isMaxFeeError ? t('editGasMaxFeeLow') : null);
+    setError(
+      isMaxPriorityFeeError || isMaxFeeError ? t('editGasTooLow') : null,
+    );
+  }, [maxPriorityFee, gasFeeEstimates, maxFee, isGasEstimatesLoading, t]);
+
+  const { currency, numberOfDecimals } = useUserPreferencedCurrency(SECONDARY);
```

Comment:
actually upon further reflection, it wouldn't make sense in the useGasFeeEsimtates hook because you need access to the *current input values*, not just the estimates. If anything you'd have to create a new hook that takes in the current values from either the selected estimate OR the input.